### PR TITLE
ON-33060 # Clears label when lookup prefills undefined on autocomplete

### PR DESCRIPTION
**Pull request by jdawg093**

### CODE REVIEW

File: src/components/renderer/AutocompleteDropdown.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ function AutocompleteDropdown<T>({
     setError(null)
     onClose()
 
-    if (!value && Array.isArray(options)) {
```
Comment (Zaxist) [inline src/components/renderer/AutocompleteDropdown.tsx:102]: Shouldn't this still only happen when there is no value?

Comment (Zaxist) [reply src/components/renderer/AutocompleteDropdown.tsx:102]: nvm

File: src/components/renderer/AutocompleteDropdown.tsx [hunk 2]
```diff
@@ -114,15 +114,16 @@ function AutocompleteDropdown<T>({
         }
```
Comment (Zaxist) [inline src/components/renderer/AutocompleteDropdown.tsx:114]: I think we probably only want the `onSelectOption` to happen if that option is not already selected.

File: src/components/renderer/AutocompleteDropdown.tsx [hunk 3]
```diff
@@ -71,6 +71,58 @@ function AutocompleteDropdown<T>({
...
+    },
+    [
+      label,
+      onChangeLabel,
+      onChangeValue,
+      onClose,
+      onSelectOption,
+      options,
+      setIsDirty,
+      value,
```
Comment (mymattcarroll) [inline src/components/renderer/AutocompleteDropdown.tsx:122]: Does this still need `value` and `label` as dependencies?

Comment (jdawg093) [reply src/components/renderer/AutocompleteDropdown.tsx:122]: It does not. Updating with the change now :)

File: src/components/renderer/AutocompleteDropdown.tsx [hunk 4]
```diff
@@ -71,6 +71,52 @@ function AutocompleteDropdown<T>({
...
+            if (!value) {
+              console.log('Setting value after blurring away from autocomplete')
+              onSelectOption(option)
+            }
+            return
+          }
+        }
+
+        console.log('Removing label after blurring away from autocomplete')
+        onChangeValue(undefined)
```
Comment (mymattcarroll) [inline src/components/renderer/AutocompleteDropdown.tsx:105]: This should not be required.

File: src/components/renderer/AutocompleteDropdown.tsx [hunk 5]
```diff
@@ -71,6 +71,52 @@ function AutocompleteDropdown<T>({
...
+      if (Array.isArray(options)) {
+        // If there is no option currently selected but the typed in label
+        // matches an option's label, set that option as the value, otherwise remove label
+        if (inputOption.label) {
+          const lowerCase = inputOption.label.toLowerCase()
+          const option = options.find(
+            (option) => option.label.toLowerCase() === lowerCase,
+          )
+          if (option) {
+            if (!value) {
```
Comment (mymattcarroll) [inline src/components/renderer/AutocompleteDropdown.tsx:96]: Why is this here?

File: src/components/renderer/AutocompleteDropdown.tsx (hunk 4 above)
Comment (mymattcarroll) [reply src/components/renderer/AutocompleteDropdown.tsx:105]: ignore

File: src/components/renderer/AutocompleteDropdown.tsx (hunk 5 above)
Comment (mymattcarroll) [reply src/components/renderer/AutocompleteDropdown.tsx:96]: ignore

File: src/components/renderer/AutocompleteDropdown.tsx [hunk 6]
```diff
@@ -71,6 +71,51 @@ function AutocompleteDropdown<T>({
...
+      if (Array.isArray(options)) {
+        // If there is no option currently selected but the typed in label
+        // matches an option's label, set that option as the value, otherwise remove label
+        if (inputOption.label) {
+          const lowerCase = inputOption.label.toLowerCase()
+          const option = options.find(
+            (option) => option.label.toLowerCase() === lowerCase,
+          )
+          if (option) {
+            if (!inputOption.value) {
```
Comment (mymattcarroll) [inline src/components/renderer/AutocompleteDropdown.tsx:96]:
```suggestion
            if (inputOption.value !== option.value) {
```

File: src/form-elements/FormElementAutocomplete.tsx [hunk 7]
```diff
@@ -103,6 +103,7 @@ const AutocompleteFilter = React.memo(function AutocompleteFilter({
     if (option && label !== option.label) {
       setLabel(option.label)
     }
+
```
Comment (mymattcarroll) [inline src/form-elements/FormElementAutocomplete.tsx:106]:
```suggestion
```
